src/test/rustdoc/intra-link-self.rs

```diff
@@ -27,3 +27,57 @@ impl Bar {
         unimplemented!()
     }
 }
+
+// @has foo/struct.WithSelf.html '//a/@href' '../foo/struct.WithSelf.html#method.basic'
+// @has foo/struct.WithSelf.html '//a/@href' '../foo/struct.WithSelf.html#method.basic_par'
+// @has foo/struct.WithSelf.html '//a/@href' '../foo/struct.WithSelf.html#method.basic_prefix'
+//
+// @has foo/struct.WithSelf.html '//a/@href' '../foo/struct.WithSelf.html#method.name_self'
+// @has foo/struct.WithSelf.html '//a/@href' '../foo/struct.WithSelf.html#method.name_self_par'
+// @has foo/struct.WithSelf.html '//a/@href' '../foo/struct.WithSelf.html#method.name_self_prefix'
+
+pub struct WithSelf;
+
+impl WithSelf {
+    /// [`basic`]
+    ///
+    /// [`basic`]: Self::basic
+    pub fn basic() -> Self {
```

Comment:
Why are you returning `Self` everytime btw?

Reply:
It's mistake, I copy/pasted the `new` method and forgot to update it.
I'll remove it, thanks!